### PR TITLE
fix: prevent BelongsTo's inverse association from itself creating a BelongsTo (#15625)

### LINKED ISSUE #15625: Multiple `belongsTo` associations in same model to same target not working with inverse
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description
When using multiple `belongsTo` associations in same model to same target with inverse, it complains that associations are not compatible.
Associations are defined from side where FK is located, with `inverse` property specified.
First associations is normally defined. But second is causing problem. Second associations works if `inverse` is removed.
I think that the problem is when defining second relation with `inverse`, it is not creating inverse relation properly, since it's comparing it with already existing 1st relation 

### Reproducible Example

Here is the link to the SSCCE for this issue: [link](https://github.com/apstanisic/sequelize-sscce)
PR: [link](https://github.com/sequelize/sequelize-sscce/pull/250)
```ts
 class Post extends Model {}
  class Author extends Model {}

  Post.init(
    {
      id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
      content: DataTypes.TEXT,
      authorId: DataTypes.INTEGER,
      coAuthorId: DataTypes.INTEGER,
    },
    {
      sequelize,
      modelName: "Post",
    }
  );

  Author.init(
    {
      id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
      name: DataTypes.TEXT,
    },
    {
      sequelize,
      modelName: "Author",
    }
  );

  Post.belongsTo(Author, {
    foreignKey: "authorId",
    as: "author",
    targetKey: "id",
    inverse: { as: "myBooks", type: "hasMany" },
  });

  Post.belongsTo(Author, {
    foreignKey: "coAuthorId",
    as: "coAuthor",
    targetKey: "id",
  // Code works if this line is removed
    inverse: { as: "notMyBooks", type: "hasMany" },
  });


```

### What do you expect to happen?

To define 2 associations with inverse associations.
### What is actually happening?

It throws an error saying that they are `incompatible because their options are not reconcilable`.
Error:
```
    [cause]: AssociationError [SequelizeAssociationError]: The association "notMyBooks" needs to define the BelongsTo association "Author" from Post to Author,
    but that child association has already been defined as BelongsTo, to Author by this call:
    
    Post.belongsTo(Author, {
      foreignKey: { name: 'authorId' },
      as: 'author',
      targetKey: 'id',
      inverse: { as: 'myBooks', type: 'hasMany' },
      hooks: false,
      name: { plural: 'authors', singular: 'author' }
    })
    
    That association would be re-used if compatible, but it is incompatible because their options are not reconcilable:
    
    Options of the association to create:
    {
      as: 'Author',
      foreignKey: { name: 'coAuthorId' },
      hooks: false,
      name: { plural: 'Authors', singular: 'Author' },
      targetKey: 'id'
    }
    
    Options of the existing association:
    {
      as: 'Author',
      foreignKey: { name: 'authorId' },
      hooks: false,
      name: { plural: 'Authors', singular: 'Author' },
      targetKey: 'id'
    }
```
### Environment

- Sequelize version: 7.0.0-alpha.20
- Node.js version: v18.13.0
- If TypeScript related: TypeScript version: 4.9.4, Not TS related
- Database & Version: postgres 15 and sqlite from sscce
- Connector library & Version: pg 8.9.0 in project, and default sqlite from sscce

## Would you be willing to resolve this issue by submitting a Pull Request?

Open to take a look, but I think this is too deep in the library.
- [ ] Yes, I have the time and I know how to start.
- [x] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [ ] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

### PR DESCRIPTION
**Pull request by ephys**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

Closes #15625

This comment explains the issue: https://github.com/sequelize/sequelize/issues/15625#issuecomment-1408092528

I've also updated `HasOneCreateAssociationMixin` to align it with `HasManyCreateAssociationMixin`. Something I noticed while writing the association documentation

### CODE REVIEW

File: packages/core/src/associations/has-one.ts [hunk 1]
```diff
@@ -97,14 +98,12 @@ export class HasOne<
     }
 
     if ('keyType' in options) {
-      throw new TypeError('Option "keyType" has been removed from the BelongsTo\'s options. Set "foreignKey.type" instead.');
+      throw new TypeError(`Option "keyType" has been removed from the BelongsTo's options. Set "foreignKey.type" instead.`);
     }
 
-    // TODO: throw is source model has a composite primary key.
```
Comment (WikiRik) [inline packages/core/src/associations/has-one.ts:103]: This is not something we want anymore?

File: packages/core/test/unit/associations/belongs-to.test.ts [hunk 2]
```diff
@@ -103,6 +104,40 @@ describe(getTestDialectTeaser('belongsTo'), () => {
...
+        inverse: { as: 'myBooks', type: 'hasMany' },
+      })
+      declare author: NonAttribute<Author>;
+
+      declare authorId: number;
+
+      @BelongsTo(() => Author, {
+        foreignKey: 'coAuthorId',
+        targetKey: 'id',
+        // Code works if this line is removed
```
Comment (WikiRik) [inline packages/core/test/unit/associations/belongs-to.test.ts:124]:
```suggestion
```

**Review by WikiRik — COMMENTED** [review]
Two small remarks

File: packages/core/src/associations/has-one.ts (hunk 1 above)
Comment (ephys) [reply packages/core/src/associations/has-one.ts:103]: It is, but belongsTo is the association that handles creating the fk so this comment is redundant

**Review by WikiRik — APPROVED** [review]

**Pull request merged**